search-service/src/test/kotlin/com/egm/stellio/search/service/AttributeInstanceServiceTests.kt

```diff
@@ -659,6 +659,41 @@ class AttributeInstanceServiceTests : WithTimescaleContainer, WithKafkaContainer
         }
     }
 
+    @Test
+    fun `it should modify attribute instance`() = runTest {
+        val attributeInstance = gimmeAttributeInstance(incomingTemporalEntityAttribute.id)
+        attributeInstanceService.create(attributeInstance)
+
+        val instanceTemporalFragment =
+            loadSampleData("fragments/temporal_instance_fragment.jsonld")
+
+        val temporalEntitiesQuery = gimmeTemporalEntitiesQuery(
+            TemporalQuery(
+                timerel = TemporalQuery.Timerel.AFTER,
+                timeAt = ZonedDateTime.parse("1970-01-01T00:00:00Z")
+            )
+        )
+        val oldInstanceResult =
+            attributeInstanceService.search(temporalEntitiesQuery, incomingTemporalEntityAttribute)
+                .shouldSucceedAndResult()
+
+        attributeInstanceService.modifyAttributeInstance(
+            entityId,
+            INCOMING_PROPERTY,
+            attributeInstance.instanceId,
+            JsonLdUtils.expandJsonLdFragment(
+                instanceTemporalFragment.deserializeAsList().first(),
+                DEFAULT_CONTEXTS
+            ) as ExpandedAttributeInstance
+        ).shouldSucceedAndResult()
+
+        val newInstanceResult =
+            attributeInstanceService.search(temporalEntitiesQuery, incomingTemporalEntityAttribute)
+                .shouldSucceedAndResult()
+
+        assertNotEquals(oldInstanceResult, newInstanceResult)
```

Comment:
or, better, check that the new instance contains exactly what you expect it to contain